project/Dependencies.scala

```diff
@@ -54,25 +55,15 @@ object Dependencies {
   val scalaCheckVersion = "1.15.1"
 
   val Versions =
-    Seq(
-      crossScalaVersions := Seq(scala212Version, scala213Version),
-      scalaVersion := {
-        System.getProperty("akka.build.scalaVersion", "default") match {
-          case v if v.startsWith("2.13") => scala213Version
-          case v if v.startsWith("2.12") => scala212Version
-          case v if v.startsWith("3.")   => scala3Version
-          case "default"                 => scala213Version
-        }
-      },
-      java8CompatVersion := {
-        CrossVersion.partialVersion(scalaVersion.value) match {
-          // java8-compat is only used in a couple of places for 2.13,
-          // it is probably possible to remove the dependency if needed.
-          case Some((3, _))            => "1.0.0"
-          case Some((2, n)) if n >= 13 => "1.0.0"
-          case _                       => "0.8.0"
-        }
-      })
+    Seq(crossScalaVersions := allScalaVersions, scalaVersion := allScalaVersions.head, java8CompatVersion := {
+      CrossVersion.partialVersion(scalaVersion.value) match {
+        // java8-compat is only used in a couple of places for 2.13,
+        // it is probably possible to remove the dependency if needed.
+        case Some((3, _))            => "1.0.0"
+        case Some((2, n)) if n >= 13 => "1.0.0"
+        case _                       => "0.8.0"
```

Comment:
if only used for 2.13 code base, why we pick 0.8.0 for 2.12? This has caused so much trouble already. 

I guess there is a reason for sticking to 0.8.0. Does anyone know it? Could we make add a comment here?

Reply:
I think the comment is meant to be read as "java8-compat is only used in a couple of places for 2.13, and in more places for 2.12".

Reply:
Ok, but why are will using scala-java8-compat  v0.8.0 for the 2.12 build. This is not clear to me and we are do this very explicitly here.

Reply:
I suspect we always used 0.8.0 for 2.12, and when we added support for 2.13 we had to update to 1.0.0 there, but didn't want to risk breaking 2.12 projects so left it at 0.8.0 for 2.12.

We could check for breaking changes between 0.8.0 and 1.0.0 and if there's nothing serious we can update this to use 1.0.0 everywhere.